src/event-db/src/types/snapshot.rs

```diff
@@ -0,0 +1,155 @@
+use crate::types::utils::serialize_systemtime_as_rfc3339;
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum SnapshotVersion {
+    Latest,
+    Number(i32),
+    Name(String),
+}
+
+impl Serialize for SnapshotVersion {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Self::Latest => "latest".serialize(serializer),
+            Self::Number(number) => number.to_string().serialize(serializer),
+            Self::Name(name) => name.serialize(serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for SnapshotVersion {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct SnapshotVersionVisitor;
+        impl<'de> serde::de::Visitor<'de> for SnapshotVersionVisitor {
+            type Value = SnapshotVersion;
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str(
+                    r#"Expect one of the following options: "latest", "25", "Fund 10" etc."#,
+                )
+            }
+            fn visit_borrowed_str<E>(self, v: &'de str) -> Result<Self::Value, E>
```

Comment:
I think this can be `visit_str` instead of `visit_borrowed_str` - we only need the `borrowed` variant if we require that the lifetime of the str is long enough for it to live in the final struct (e.g. if we had a struct like `struct Foo<'a> { inner: &'a str }`.

But since we don't have that, we can use `visit_str` which has a lifetime which is only guaranteed to last for the body of the function, which is enough for our needs

Reply:
fixed